rtbkit/common/augmentor_interface.cc

```diff
@@ -0,0 +1,86 @@
+/* augmentor_interface.cc
+   Mathieu Stefani, 20 janvier 2016
+   Copyright (c) 2015 Datacratic.  All rights reserved.
+   
+*/
+
+#include "augmentor_interface.h"
+
+using namespace Datacratic;
+
+namespace RTBKIT {
+
+AugmentorInterface::AugmentorInterface(
+    ServiceBase& parent, const std::string& serviceName)
+  : ServiceBase(serviceName, parent),
+    inbox(65536)
+{ }
+
+AugmentorInterface::AugmentorInterface(
+    std::shared_ptr<ServiceProxies> proxies,
+    const std::string& serviceName)
+  : ServiceBase(serviceName, std::move(proxies)),
+    inbox(65536)
+{ }
+
+void AugmentorInterface::init(){
+    inbox.onEvent = [&] (AugmentMessage&& message)
+        {
+            doSendAugmentMessage(
+                message.instance,
+                std::move(message.augmentorName),
+                message.auction,
+                message.agents,
+                std::move(message.date));
+        };
+
+    addSource("AugmentorInterface::inbox", inbox);
+}
+
+void AugmentorInterface::sendAugmentMessage(
+            const std::shared_ptr<AugmentorInstanceInfo>& instance,
+            const std::string& augmentorName,
+            const std::shared_ptr<Auction>& auction,
+            const std::set<std::string>& agents,
+            Datacratic::Date date){
+
+    AugmentMessage m;
+    m.instance = instance;
+    m.augmentorName = std::move(augmentorName);
```

Comment:
You've been beaten by the gods of C++ here. Basically, even though the `std::move()` expression is saying that it will move the string, it won't as augmentorName is `const` and you can not move from an `const rvalue`. It's minor but if you want to fix that:

1/ either remove the `move()` as it will just copy
2/ Pass the `augmentorName` by simple value and keep the move.